File: pset6/readability.py
from cs50 import get_string

# sets the values to 0


letters = 0
words = 1
sentences = 0

# gets text from user


s = get_string("Text: ")

# calculates length of the text


length = len(s)

# considers bad scenarios


if s[0] == " ":
    words = words - 1

if s[length - 1] == " ":
    words = words - 1

for i in range(length):
    if s[i].isalpha() == True:
        letters += 1
    # counts words
    if s[i] == " " and s[i - 1] != " ":
        words += 1
    # counts sentences
    if s[i] == "." or s[i] == "?" or s[i] == "!":
        sentences += 1

# calculates index and grade


L = float(letters) / float(words) * 100
S = float(sentences) / float(words) * 100
index = 0.0588 * L - 0.296 * S - 15.8
grade = round(index)


# determines what to print as output


if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")

File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

//setting variable counters
int letters = 0;
int words = 1;
int sentences = 0;
int main(void)
{
    //prompts user for text input
    string s = get_string("Text: \n");
    //stores length of the string input
    int length = strlen(s);

    //takes care of the case where first character is a space
    if (s[0] == 32)
    {
        words = words - 1;
    }
    //takes care of the case where the last character is a space
    if (s[length - 1] == 32)
    {
        words = words - 1;
    }

    //iterates through the characters in the string
    for (int i = 0; i < length; i++)
    {
        //counts letters
        if ((s[i] >= 65 && s[i] <= 90) || (s[i] >= 97 && s[i] <= 122))
        {
            letters++;
        }
        //counts words
        if (s[i] == 32 && s[i - 1] != 32)
        {
            words++;
        }
        //counts sentences
        if ((s[i] == 46 || s[i] == 33 || s[i] == 63))
        {
            sentences++;
        }

    }



    //calculations for index
    double L = (double) letters / (double) words * 100;
    double S = (double) sentences / (double) words * 100;
    double index =  0.0588 * (double) L - 0.296 * (double) S - 15.8;
    int grade = round(index);

    //output
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

}
